// @vue-macros/named-template/dist/chunk-WIBDHOGO.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});






var _chunkQ5JZ3HH6js = require('./chunk-Q5JZ3HH6.js');

// src/index.ts
var _unplugin = require('unplugin');




var _common = require('@vue-macros/common');
function resolveOption(options) {
  const version = options.version || _common.detectVueVersion.call(void 0, );
  return {
    include: [_common.REGEX_VUE_SFC],
    ...options,
    version
  };
}
var name = "unplugin-vue-named-template";
var PrePlugin = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}) => {
    const options = resolveOption(userOptions);
    const filter = _common.createFilter.call(void 0, options);
    const templateContent = {};
    return {
      name: `${name}-pre`,
      enforce: "pre",
      loadInclude(id) {
        return id.includes(_chunkQ5JZ3HH6js.QUERY_TEMPLATE);
      },
      load(id) {
        var _a;
        const { filename, query } = _chunkQ5JZ3HH6js.parseVueRequest.call(void 0, id);
        const content = (_a = templateContent[filename]) == null ? void 0 : _a["mainTemplate" in query ? _chunkQ5JZ3HH6js.MAIN_TEMPLATE : query.name];
        return content;
      },
      transformInclude(id) {
        return filter(id) || id.includes(_chunkQ5JZ3HH6js.QUERY_NAMED_TEMPLATE);
      },
      transform(code, id) {
        if (id.includes(_chunkQ5JZ3HH6js.QUERY_NAMED_TEMPLATE)) {
          const { filename, query } = _chunkQ5JZ3HH6js.parseVueRequest.call(void 0, id);
          const { name: name2 } = query;
          const request = `${filename}?vue&${_chunkQ5JZ3HH6js.QUERY_TEMPLATE}&name=${name2}`;
          return `import { createTextVNode } from 'vue'
          import { render } from ${JSON.stringify(request)}
export default {
  render: (...args) => {
    const r = render(...args)
    return typeof r === 'string' ? createTextVNode(r) : r
  }
}`;
        } else {
          return _chunkQ5JZ3HH6js.preTransform.call(void 0, code, id, templateContent);
        }
      }
    };
  }
);
var PostPlugin = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}) => {
    const options = resolveOption(userOptions);
    const filter = _common.createFilter.call(void 0, options);
    const customBlocks = {};
    function transformInclude(id) {
      return filter(id) || id.includes(_chunkQ5JZ3HH6js.QUERY_TEMPLATE);
    }
    return {
      name: `${name}-post`,
      enforce: "post",
      transformInclude,
      transform(code, id) {
        return _chunkQ5JZ3HH6js.postTransform.call(void 0, code, id, customBlocks);
      },
      rollup: {
        transform: {
          order: "post",
          handler(code, id) {
            if (!transformInclude(id))
              return;
            return _chunkQ5JZ3HH6js.postTransform.call(void 0, code, id, customBlocks);
          }
        }
      }
    };
  }
);
var plugin = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}, meta) => {
    return [PrePlugin.raw(userOptions, meta), PostPlugin.raw(userOptions, meta)];
  }
);
var src_default = plugin;





exports.PrePlugin = PrePlugin; exports.PostPlugin = PostPlugin; exports.src_default = src_default;
